### ISSUE TITLE
Warning: $e not defined

### ISSUE THREAD

**Issue by falkgeist** [terminal42/contao-notification_center#294]
In https://github.com/terminal42/contao-notification_center/blob/3aab8f545db43e5bbce1771ec9bdfe2ced2c7d50/library/NotificationCenter/Gateway/Email.php#L167C174-L167C174 you reference the exception from a few lines before, but there might be case (however rare), where the $blnSet might be false without the exception being caught and there not being defined.

This is only generating a warning, not an error, and in my implementation I caught the resulting error with a Message ("notification could not be sent"), but this is not necessarily expected behavior, even for debug mode.

I got to say though, I forced the warning in a way, since my code didn't pass an email. But since https://github.com/contao/contao/blob/a0cc1ef31e206c61abc7380f03f0ee7018d810fb/core-bundle/contao/library/Contao/Email.php#L398 can also just return false, if theres no mails passed, that case should optimally be covered in above referenced method.

**Comment by fritzmg** [comment]
It could be changed to something like this:

```diff
+   $errorMessage = 'N/A';

    try {
        $blnSent = $objEmail->sendTo($objDraft->getRecipientEmails());
    } catch (\Exception $e) {
        $blnSent = false;
+       $errorMessage = $e->getMessage();
    }

    if (!$blnSent) {
-       \System::log(sprintf('Could not send email to "%s" for message ID %s: %s', implode(', ', $objDraft->getRecipientEmails()), $objDraft->getMessage()->id, $e->getMessage()), __METHOD__, TL_ERROR);
+       \System::log(sprintf('Could not send email to "%s" for message ID %s: %s', implode(', ', $objDraft->getRecipientEmails()), $objDraft->getMessage()->id, $errorMessage), __METHOD__, TL_ERROR);
    }
```

@falkgeist can you test these changes and provide a PR?

**Comment by aschempp** [comment]
fixed in 73f3c4391bfb2b569611eff137b5bcfad80d5781
